src/pages/api/productDateInformation.ts

```diff
@@ -0,0 +1,169 @@
+
+export interface ProductDateInformation {
+    startDateDay: string;
+    startDateMonth: string;
+    startDateYear: string;
+    endDateDay: string;
+    endDateMonth: string;
+    endDateYear: string;
+}
+
+export const dateValidationSchema = yup.object({
+    startDateDay: yup.string().notRequired(),
+    startDateMonth: yup.string().notRequired(),
+    startDateYear: yup.string().notRequired(),
+    endDateDay: yup.string().notRequired(),
+    endDateMonth: yup.string().notRequired(),
+    endDateYear: yup.string().notRequired(),
+});
+
+export const combinedDateSchema = yup.object({
+    endDate: yup.date().min(yup.ref('startDate'), 'The end date must be after the start date'),
+});
+
+export const getErrorIdFromDateError = (errorPath: string): string => {
+    switch (errorPath) {
+        case 'startDateDay':
+            return 'start-date';
+        case 'startDateMonth':
+            return 'start-date';
+        case 'startDateYear':
+            return 'start-date';
+        case 'endDateDay':
+            return 'end-date';
+        case 'endDateMonth':
+            return 'end-date';
+        case 'endDateYear':
+            return 'end-date';
+        case 'startDate':
+            return 'start-date';
+        case 'endDate':
+            return 'end-date';
+        default:
+            throw new Error(`Could not match the following error with an expected input. Error path: ${errorPath}.`);
+    }
+};
+
+const isDatesFieldEmpty = (day: string, month: string, year: string): boolean =>
+    day === '' && month === '' && year === '';
+
+export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
+    try {
+        let errors: ErrorInfo[] = [];
+
+        const {
+            startDateDay,
+            startDateMonth,
+            startDateYear,
+            endDateDay,
+            endDateMonth,
+            endDateYear,
+            productDates,
+        } = req.body;
+
+        const dateInput: ProductDateInformation = {
+            startDateDay,
+            startDateMonth,
+            startDateYear,
+            endDateDay,
+            endDateMonth,
+            endDateYear,
+        };
+
+        if (!productDates) {
+            errors.push({ errorMessage: 'Choose one of the options below', id: 'product-dates-required' });
+            updateSessionAttribute(req, PRODUCT_DATE_ATTRIBUTE, { errors, dates: dateInput });
+            redirectTo(res, '/productDateInformation');
+            return;
+        }
+
+        let startDate = moment();
+        let endDate = moment().add(100, 'y');
+
+        if (productDates === 'Yes') {
+            const isStartDateEmpty = isDatesFieldEmpty(startDateDay, startDateMonth, startDateYear);
+            const isEndDateEmpty = isDatesFieldEmpty(endDateDay, endDateMonth, endDateYear);
+
+            if (isStartDateEmpty && isEndDateEmpty) {
+                errors.push({ errorMessage: 'Enter a start or end date', id: 'start-date' });
+            }
```

Comment:
When the user enters no start or end date, should we render this error on both of the inputs?

Reply:
yeah have made this change
![Uploading Screenshot 2020-09-22 at 14.07.21.png…]()